docs/safe_lambda_deployments.rst

```diff
@@ -102,35 +103,43 @@ resource:
           PreTraffic: !Ref PreTrafficLambdaFunction
           PostTraffic: !Ref PostTrafficLambdaFunction
 
+  PreTrafficLambdaFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      Handler: preTrafficHook.handler
+      Role: [arn To IAM Role With Code DeployAndInvoke Permissions]
+      Runtime: nodejs6.10
+      Tracing: Active
+      Timeout: 10
+      FunctionName: 'CodeDeployHook_preTrafficHook'
+      DeploymentPreference:
+        Enabled: false
+      Environment:
+        Variables:
+          CurrentVersion: !Ref MyLambdaFunction.Version
+
 When you update your function code and deploy the SAM template using
 CloudFormation, the following happens:
 
--  A new Lambda Version is published from the new code
--  Since a deployment preference is set, CodeDeploy takes over the job
-   of actually shifting traffic from old version to new version.
--  Before traffic shifting starts, CodeDeploy will invoke the
-   **PreTraffic Hook** Lambda Function. This Lambda function must call
-   back to CodeDeploy with explicit Success or Failure. On Failure, it
-   will abort and report a failure back to CloudFormation. On Success,
-   CodeDeploy will proceed to traffic shifting.
--  ``Type: Linear10PercentEvery10Minutes`` instructs CodeDeploy to start with
-   10% traffic on new version and add 10% every 10 minutes. It will complete traffic shifting in 100 minutes. 
--  During traffic shifting, if any of the CloudWatch Alarms go to
-   *Alarm* state, CodeDeploy will immediately flip the Alias back to old
-   version and report a failure to CloudFormation.
--  After traffic shifting completes, CodeDeploy will invoke the
-   **PostTraffic Hook** Lambda Function. This is similar to PreTraffic
-   Hook where the function must callback to CodeDeploy to report a
-   Success or Failure. PostTraffic hook is a great place to run
-   integration tests or other validation actions.
--  If everything went well, the Alias will be pointing to the new Lambda
-   Version.
+- Cloudformation pubilshes a new Lambda Version from the new code
+- Since a deployment preference is set, CodeDeploy takes over the job of actually shifting traffic from old version to new version.
+- Before traffic shifting starts, CodeDeploy will invoke the **PreTraffic Hook** Lambda Function. This Lambda function must call back to CodeDeploy with an explicit status of Success or Failure, via the PutLifecycleEventHookExecutionStatus_ API. On Failure, CodeDeploy will abort and report a failure back to CloudFormation. On Success, CodeDeploy will proceed with the specified traffic shifting. Here_ is a sample Lambda Hook function.
```

Comment:
Can you add a note here about making sure the version of the SDK has the `PutLifecycleEventHookExecutionStatus` API. Under the Pre and Post hook section is probably the best place for this data. It would be nice to call out the min version of SDK for Java, Python (version 1.4.8), and Node.

Reply:
Note added